inc/section.class.php

```diff
@@ -383,4 +383,49 @@ public function showSubForm($ID) {
       echo '</table>';
       Html::closeForm();
    }
+
+   /**
+    * Get either:
+    *  - section and questions of the target parent form
+    *  - questions of target section
+    *
+    * @param int $parent target parent form
+    * @param int $id target section
+    * @return array
+    */
+   public static function getFullData($parent, $id = null) {
+      global $DB;
+
+      if ($parent) {
+         $data = [];
+         $data['_sections'] = iterator_to_array($DB->request([
+            'FROM' => \PluginFormcreatorSection::getTable(),
+            'WHERE' => ["plugin_formcreator_forms_id" => $parent]
+         ]));
+
+         $ids = [];
+         foreach ($data['_sections'] as $section) {
+            $ids[] = $section['id'];
+         }
+
+         $data = $data + \PluginFormcreatorQuestion::getFullData($ids);
+      } else {
+         if ($id == null) {
+            throw new \InvalidArgumentException(
+               "Parameter 'id' can't be null if parameter 'parent' is not specified"
```

Comment:
Same as above : if you throw an exception, you need to have a try / catch in one of the caller methods